android/engine/src/test/java/org/smartregister/fhircore/engine/util/helper/TransformSupportServicesTest.kt

```diff
@@ -57,4 +63,81 @@ class TransformSupportServicesTest : RobolectricTest() {
         Immunization.ImmunizationReactionComponent
     )
   }
+
+  @Test
+  fun `createType() should return Diagnosis when given EpisodeOfCare_Diagnosis`() {
+    Assert.assertTrue(
+      transformSupportServices.createType("", "EpisodeOfCare_Diagnosis") is
+        EpisodeOfCare.DiagnosisComponent
+    )
+  }
+
+  @Test
+  fun `createType() should return Diagnosis when given Encounter_Diagnosis`() {
+    Assert.assertTrue(
+      transformSupportServices.createType("", "Encounter_Diagnosis") is Encounter.DiagnosisComponent
+    )
+  }
+
+  @Test
+  fun `createType() should return EncounterParticipant when given Encounter_Participant`() {
+    Assert.assertTrue(
+      transformSupportServices.createType("", "Encounter_Participant") is
+        Encounter.EncounterParticipantComponent
+    )
+  }
+
+  @Test
+  fun `createType() should return CarePlanActivity when given CarePlan_Activity`() {
+    Assert.assertTrue(
+      transformSupportServices.createType("", "CarePlan_Activity") is
+        CarePlan.CarePlanActivityComponent
+    )
+  }
+
+  @Test
+  fun `createType() should return CarePlanActivityDetail when given CarePlan_ActivityDetail`() {
+    Assert.assertTrue(
+      transformSupportServices.createType("", "CarePlan_ActivityDetail") is
+        CarePlan.CarePlanActivityDetailComponent
+    )
+  }
+
+  @Test
+  fun `createType() should return PatientLink when given Patient_Link`() {
+    Assert.assertTrue(
+      transformSupportServices.createType("", "Patient_Link") is Patient.PatientLinkComponent
+    )
+  }
+
+  @Test
+  fun `createType() should return Time when given time`() {
+    Assert.assertTrue(transformSupportServices.createType("", "time") is TimeType)
+  }
+
+  @Test
+  fun `createResource() should add resource into output when given Patient and atRootOfTransForm as True`() {
+    transformSupportServices.createResource("", Patient(), true)
```

Comment:
Can we assert that the output size is 0 before invoking `createResource`?

Reply:
Sure